includes/data-stores/class-wc-coupon-data-store-cpt.php

```diff
@@ -210,6 +210,9 @@ private function update_post_meta( &$coupon ) {
 				case 'email_restrictions' :
 					$updated = update_post_meta( $coupon->get_id(), $meta_key, array_filter( array_map( 'sanitize_email', $value ) ) );
 					break;
+				case 'date_expires' :
+					$updated = update_post_meta( $coupon->get_id(), $meta_key, ( $value ? $value->date( 'Y-m-d' ) : '' ) );
```

Comment:
@claudiulodro If the date stored is local time, we should introduce a new, 2.7.x meta field which stores as a UTC timestamp.

Both get saved for backwards compatibility, but we rely on the new UTC timestamp if it exists since that is more reliable.